### PR TITLE
mc admin config set/get changes

### PR DESCRIPTION
**Pull request by harshavardhana**

### CODE REVIEW

File: cmd/admin-config-del.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+USAGE:
+  {{.HelpName}} TARGET
+
+FLAGS:
+  {{range .VisibleFlags}}{{.}}
+  {{end}}
+EXAMPLES:
+  1. Enable worm mode on MinIO server/cluster.
+     $ {{.HelpName}} myminio/ notify_mqtt:target1
```
Comment (kannappanr) [inline cmd/admin-config-del.go]: comment and the command doesn't match

Comment (harshavardhana) [reply cmd/admin-config-del.go]: Oh yes typo

File: cmd/admin-config-get.go [hunk 2]
```diff
@@ -38,23 +40,21 @@ FLAGS:
   {{range .VisibleFlags}}{{.}}
   {{end}}
 EXAMPLES:
-  1. Get server configuration of a MinIO server/cluster.
-     {{.Prompt}} {{.HelpName}} play/
+  1. Get the current status of 'worm' mode on MinIO server/cluster.
+     $ {{.HelpName}} play/ worm
+     state="off"
```
Comment (Praveenrajmani) [inline cmd/admin-config-get.go]: can we add some more examples on querying the subsystems?

Comment (harshavardhana) [reply cmd/admin-config-get.go]: Definitely its not done yet @Praveenrajmani

File: cmd/admin-config-del.go [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+	// Print the general set config status
+	if u.delConfigStatus {
+		msg += console.Colorize("DelConfigSuccess",
+			"Deleting key has been successful.\n")
+		suggestion := fmt.Sprintf("mc admin service restart %s", u.targetAlias)
+		msg += console.Colorize("DelConfigSuccess",
+			fmt.Sprintf("Please restart your server with `%s`.\n", suggestion))
+	} else {
+		msg += console.Colorize("DelConfigFailure",
+			"Delting new MinIO configuration file has failed.\n")
```
Comment (poornas) [inline cmd/admin-config-del.go]:
```suggestion
 			"Deleting new MinIO configuration file has failed.\n")
```

**Comment by poornas** [comment]
```
 mc admin config history myminio
mc: <ERROR> `myminio` is not a mc command. See `mc --help`. 
```
would be nice if this shows help for history

**Comment by harshavardhana** [comment]
This is not true for almost all sub-commands such as `mc config host myminio`  - if there are complaints for that too then we should fix it separately not in this PR.

These commands have sub-commands so `mc admin config history -h` will give you help rest will be handled by the cli package and there is no way to handle it since history itself has sub-commands of it's own. Before the control comes to us - I would perhaps enhance CLI package to print a context sensitive error by saying please type `mc admin config history --help` for more usage.

This needs a larger change which needs to be done across other sub-sub-commands style commands as well.

**Comment by harshavardhana** [comment]
Another set of changes are in as well https://github.com/minio/mc/pull/2940

**Comment by harshavardhana** [comment]
This PR can be tested with https://github.com/minio/minio/pull/8439

**Comment by poornas** [comment]
```
mc admin config set myminio region name="us-west-1" state="on"
region name=us-west-1 state=on
mc: <ERROR> Cannot set server configuration file. ParseBool: parsing '': invalid syntax.
```
with latest commit

**Comment by harshavardhana** [comment]
> mc admin config set myminio region name="us-west-1" state="on"
> region name=us-west-1 state=on
> mc: <ERROR> Cannot set server configuration file. ParseBool: parsing '': invalid syn

Looks like you didn't compile `mc` correctly @poornas 

```
~ mc admin config set myminio region name="us-west-1" state="on"
Setting new key has been successful.
Please restart your server with `mc admin service restart myminio`.
```

**Comment by poornas** [comment]
> mc admin config set myminio region name="us-west-1" state="on"





> > mc admin config set myminio region name="us-west-1" state="on"
> > region name=us-west-1 state=on
> > mc:  Cannot set server configuration file. ParseBool: parsing '': invalid syn
> 
> Looks like you didn't compile `mc` correctly @poornas
> 
> ```
> ~ mc admin config set myminio region name="us-west-1" state="on"
> Setting new key has been successful.
> Please restart your server with `mc admin service restart myminio`.
> ```

there was some invalid config in my config.json,couldn't repro when I started afresh

File: cmd/admin-config-del.go [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+	// Print the general set config status
+	if u.delConfigStatus {
+		msg += console.Colorize("DelConfigSuccess",
+			"Deleting key has been successful.\n")
+		suggestion := fmt.Sprintf("mc admin service restart %s", u.targetAlias)
+		msg += console.Colorize("DelConfigSuccess",
+			fmt.Sprintf("Please restart your server with `%s`.\n", suggestion))
+	} else {
+		msg += console.Colorize("DelConfigFailure",
+			"Deleting new MinIO configuration file has failed.\n")
```
Comment (kannappanr) [inline cmd/admin-config-del.go]:
```suggestion
			"Deleting key failed.\n")
```

File: cmd/admin-config-set.go [hunk 5]
```diff
@@ -109,13 +102,19 @@ func mainAdminConfigSet(ctx *cli.Context) error {
 	client, err := newAdminClient(aliasedURL)
 	fatalIf(err, "Unable to initialize admin connection.")
 
-	// Call get config API
-	fatalIf(probe.NewError(client.SetConfig(os.Stdin)), "Cannot set server configuration file.")
+	// Call set config API
+	input := strings.Join(args.Tail(), " ")
+	if len(input) == 0 {
+		b, e := ioutil.ReadAll(os.Stdin)
+		fatalIf(probe.NewError(e), "Cannot read from the stdIn the new settings.")
```
Comment (kannappanr) [inline cmd/admin-config-set.go]:
```suggestion
		fatalIf(probe.NewError(e), "Cannot read from the new settings from stdin.")
```

File: cmd/admin-config-get.go [hunk 6]
```diff
@@ -86,16 +98,12 @@ func mainAdminConfigGet(ctx *cli.Context) error {
...
 
 	// Call get config API
-	c, e := client.GetConfig()
-	fatalIf(probe.NewError(e), "Cannot get server configuration file.")
-
-	config := map[string]interface{}{}
-	e = json.Unmarshal(c, &config)
-	fatalIf(probe.NewError(e), "Cannot unmarshal server configuration file.")
+	buf, e := client.GetConfigKV(strings.Join(args.Tail(), " "))
+	fatalIf(probe.NewError(e), "Cannot get server config.fg")
```
Comment (poornas) [inline cmd/admin-config-get.go]:
```suggestion
 	fatalIf(probe.NewError(e), "Cannot get server config")
```

File: cmd/admin-config-help.go [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+var helpFlags = []cli.Flag{
+	cli.BoolFlag{
+		Name:  "env",
+		Usage: "list all the env only help",
+	},
+}
+
+var adminConfigHelpCmd = cli.Command{
+	Name:   "help",
+	Usage:  "help returns help for each sub-system",
```
Comment (poornas) [inline cmd/admin-config-help.go]:
```suggestion
 	Usage:  "show help for each sub-system",
```

**Pull request merged**